assets/js/blocks/checkout/inner-blocks/checkout-terms-block/test/edit.js

```diff
@@ -7,7 +7,31 @@ import { render, queryByText } from '@testing-library/react';
  * Internal dependencies
  */
 import { Edit } from '../edit';
-const blockSettingsMock = jest.requireMock( '@woocommerce/block-settings' );
+
+jest.mock( '@wordpress/data', () => ( {
+	...jest.requireActual( '@wordpress/data' ),
+	useSelect: jest.fn().mockImplementation( ( fn ) => {
```

Comment:
This was a bit tricky, I had to mock a few useSelect cases to avoid exceptions in underlying RichText component.